kedro/io/core.py

```diff
@@ -227,32 +229,23 @@ def save(self, data: _DI) -> None:
             message = f"Failed while saving data to data set {str(self)}.\n{str(exc)}"
             raise DatasetError(message) from exc
 
-    def __str__(self) -> str:
```

Comment:
IIUC, we're removing the `__str__` format right? Shouldn't we keep it around for backwards compatibility?

Reply:
If `__str__` is not defined the default is to use the `__repr__`, so it shouldn't be a problem.

![Uploading Screenshot 2024-07-09 at 14.20.15.png…]()

Reply:
Oh right. Still, we're changing __str__ right? Not sure if someone depends on str(dataset). it looks like a small, preventable breaking change.

Reply:
So you mean keeping the old `__str__`?

Reply:
Yes, at least until 0.20...